tests/unit/core/server/request/test-class-core-update-account-request.php

```diff
@@ -0,0 +1,79 @@
+			'business_name'                   => 'test business_name',
+			'business_url'                    => 'test business_url',
+			'business_support_address'        => 'test business_support_address',
+			'business_support_email'          => 'test business_support_email',
+			'business_support_phone'          => 'test business_support_phone',
+			'branding_logo'                   => 'test branding_logo',
+			'branding_icon'                   => 'test branding_icon',
+			'branding_primary_color'          => 'test branding_primary_color',
+			'branding_secondary_color'        => 'test branding_secondary_color',
+			'deposit_schedule_interval'       => 'test deposit_schedule_interval',
+			'deposit_schedule_weekly_anchor'  => 'test deposit_schedule_weekly_anchor',
+			'deposit_schedule_monthly_anchor' => 'test deposit_schedule_monthly_anchor',
+			'locale'                          => 'test locale',
+		];
+
+		$request = Update_Account::from_account_settings( $account_setting );
+
+		// Assert request class.
+		$this->assertInstanceOf( Update_Account::class, $request );
+
+		// Assert params.
+		$params = $request->get_params();
+		$this->assertSame( 'test statement_descriptor', $params['statement_descriptor'] );
+		$this->assertSame( 'test business_name', $params['business_name'] );
+		$this->assertSame( 'test business_url', $params['business_url'] );
+		$this->assertSame( 'test business_support_address', $params['business_support_address'] );
+		$this->assertSame( 'test business_support_email', $params['business_support_email'] );
+		$this->assertSame( 'test business_support_phone', $params['business_support_phone'] );
+		$this->assertSame( 'test branding_logo', $params['branding_logo'] );
+		$this->assertSame( 'test branding_icon', $params['branding_icon'] );
+		$this->assertSame( 'test branding_primary_color', $params['branding_primary_color'] );
+		$this->assertSame( 'test branding_secondary_color', $params['branding_secondary_color'] );
+		$this->assertSame( 'test deposit_schedule_interval', $params['deposit_schedule_interval'] );
+		$this->assertSame( 'test deposit_schedule_weekly_anchor', $params['deposit_schedule_weekly_anchor'] );
+		$this->assertSame( 'test deposit_schedule_monthly_anchor', $params['deposit_schedule_monthly_anchor'] );
+		$this->assertSame( 'test locale', $params['locale'] );
+
+		// Assert method and endpoint.
+		$this->assertSame( 'POST', $request->get_method() );
+		$this->assertSame( 'accounts', $request->get_api() );
+	}
+
+	public function test_from_account_settings_with_empty_input_will_throw_exception() {
+		$this->expectException( Invalid_Request_Parameter_Exception::class );
+		$this->expectExceptionMessage( 'No account settings provided' );
+
+		Update_Account::from_account_settings( [] );
+	}
+
+	public function test_from_account_settings_with_non_existing_param_continue_to_run() {
+		$none_existing_param = 'none_existing_param';
+		$this->expectException( Invalid_Request_Parameter_Exception::class );
```

Comment:
Nitpick: Move this assert exceptions after calling `Update_Account::from_account_settings` to be sure that exception will be thrown when `$request->get_param( $none_existing_param )` is called

Reply:
Good point. Addressed 03ee2e8542347e6dcb3f4c60e19e3d05a68d6541